### DISCUSSION TITLE
Carthage

### DISCUSSION THREAD

**Discussion by jimmyhoran** [MobileNativeFoundation/discussions#27] [Proposals]
I would love to see tools like Carthage moved to a foundation like this over time. Could be a good fit, assuming this foundation has the community backing to help maintain an OSS tool like Carthage.

**Comment by freak4pc** [comment]
I'm not sure yet what this foundation _is_ yet, as it's still evolving, but it sounds counter-intuitive to host any projects in it. 

It seems like this is more of a place to share common ground and experiences from the day-to-day work and challenges of mobile teams, and not necessarily endorse or recommend one specific project "by the foundation". 

Anyways, Carthage specifically already has its own well-established GH Organization and I imagine they won't benefit much from transitioning to a different one.

**Comment by keith** [comment]
What projects make the most sense to be part of the foundation is still something we're working on defining (more on that here later https://github.com/MobileNativeFoundation/discussions/discussions/21). I think like @freak4pc said, since Carthage has a well established setup, I don't think moving it to the MNF would directly provide any benefit for the users or maintainers. Currently the MNF is not providing funding to projects, so there wouldn't be any immediate influx of maintainers. I think that equation changes for other types of projects that live somewhere with no maintainers, and no one can access them either.

**Comment by thedavidharris** [comment]
Along with what's mentioned, Carthage is in a bit of a sore spot since it relies on a present xcodeproj to build the framework, and as Swift Package Manager supplants it without relying on an xcodeproj at all Carthage is falling out of date. Most attempts at trying to fix this gap relied on the `swift package generate-xcodeproj` which is deprecated, and extrapolating that I think it's fair to say that we're quickly approaching the xcodeproj itself being for non-iOS/etc. actual application bundles for 99% of package development (special use cases like cross-language things aside). Many frameworks being shipped already don't include it.

**Comment by maeselaqun** [comment]
Carthage moving to a foundation like the Swift Server Working Group or a dedicated organization would be a huge win for stability. The main hurdle is whether there's enough active interest to staff the maintenance. Many devs have already jumped ship to Swift Package Manager, so the "community backing" might be thinner than it was a few years ago. If you're tired of the basic options, you should definitely check out the sites that actually specialize in esports because they offer way more detailed bet apps [telecomasia.net/in/best-esports-betting-sites/](https://www.telecomasia.net/in/best-esports-betting-sites/) prop markets for every single map. Still, for the legacy projects that rely on it, formalizing the stewardship would stop the constant fear of it becoming completely abandoned. It’s a great idea in theory—the challenge is finding the maintainers willing to do the thankless work.
